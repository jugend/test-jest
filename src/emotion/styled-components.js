import React from 'react'
import styled, { css } from 'react-emotion'

export const BlueButton = styled('button')`
  color: blue;
`

export const BorderedButton = styled.button((props) => `
  border: 1px solid ${props.borderColor}
`)

// export const BorderedButton = styled.button((props) => ({
//   border: `1px solid ${props.borderColor}`
// }))
