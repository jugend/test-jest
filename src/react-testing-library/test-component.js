import React, { Component } from 'react'

export default class TestComponent extends Component {
  constructor() {
    super()
    this.state = { count: 0 }
  }

  render() {
    const { count } = this.state
    return (
      <button onClick={() => this.setState({ count: count + 1 })}>
        Click to increase: {count}
      </button>
    )
  }
}
