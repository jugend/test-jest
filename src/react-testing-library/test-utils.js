// test-utils.js
import React from 'react'
// Best Practice: To override and add additional helper methods
import { render } from 'react-testing-library'
// import {ThemeProvider} from 'my-ui-lib'
// import {TranslationProvider} from 'my-i18n-lib'
// import defaultStrings from 'i18n/en-x-default'

export const customRender = (node, ...options) => {
  return render(
    // <ThemeProvider theme="light">
    //   <TranslationProvider messages={defaultStrings}>
    <div data-testid="custom-render">Custom: {node}</div>,
    //   </TranslationProvider>
    // </ThemeProvider>,
    ...options,
  )
}

// re-export everything
export * from 'react-testing-library'

// override render method
// export { customRender as render }
