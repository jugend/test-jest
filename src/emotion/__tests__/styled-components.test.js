import React from 'react'
import { render } from 'react-testing-library'
import { BlueButton, BorderedButton } from '../styled-components'

it('should render component with css', () => {
  const { container, getByText, debug } = render(<BlueButton>Blue Button</BlueButton>)
  expect(container).toMatchSnapshot()
})

it('should render button with dynamic prop', () => {
  const { container, debug } = render(<BorderedButton borderColor='cyan'>Bordered Button</BorderedButton>)
  // debug()
  expect(container).toMatchSnapshot()
})
