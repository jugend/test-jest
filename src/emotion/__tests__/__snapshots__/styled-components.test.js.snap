// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button with dynamic prop 1`] = `
.emotion-0 {
  border: 1px solid cyan;
}

<div>
  <button
    class="emotion-0 emotion-1"
  >
    Bordered Button
  </button>
</div>
`;

exports[`should render component with css 1`] = `
.emotion-0 {
  color: blue;
}

<div>
  <button
    class="emotion-0 emotion-1"
  >
    Blue Button
  </button>
</div>
`;
